src/kOS/Execution/Stack.cs

```diff
@@ -48,9 +48,13 @@ private void ThrowIfInvalid(object item)
         /// <returns></returns>
         private object ProcessItem(object item)
         {
-            if (item is float)
+            if (item is float) {
                 // promote floats to doubles
-                return Convert.ToDouble(item);
+                item = Convert.ToDouble (item);
+            }
+            if (item is double && (double)item == Math.Truncate((double)item)) {
+                return Convert.ToInt32 (item); // or should it just be (int)?
```

Comment:
> only performs the rounding once

1. Does `Convert.ToInt32()` work faster if it receives round number?
2. `return Convert.ToInt32 (item);` in original code executes if and only if `item` is exactly equal it `Math.Truncate((double)item)` so in this particular situation it shouldn't matter.